massa-execution/src/worker.rs

```diff
@@ -255,106 +253,137 @@ impl ExecutionWorker {
     /// called when the blockclique changes
     fn blockclique_changed(
         &mut self,
-        mut blockclique: BlockHashMap<Block>,
-        mut finalized_blocks: BlockHashMap<Block>,
+        blockclique: BlockHashMap<Block>,
+        finalized_blocks: BlockHashMap<Block>,
     ) -> Result<(), ExecutionError> {
-        // filter out any already-finalized blocks
-        // this can happen for example on bootstrap because the consensus vs execution bootstrap snapshots may be out of sync
-        blockclique.retain(|_b_id, b| b.header.content.slot > self.last_final_slot);
-        finalized_blocks.retain(|_b_id, b| b.header.content.slot > self.last_final_slot);
-
         // stop the current VM execution and reset state to final
         // TODO make something more iterative/conservative in the future to reuse unaffected executions
         self.reset_to_final();
+        self.last_active_slot = self.last_final_slot;
 
-        // stop the current VM execution and reset state to final
-        // TODO reset VM
-
-        // gather pending finalized CSS
-        let mut css_final_blocks: Vec<(BlockId, Block)> = self
-            .ordered_pending_css_final_blocks
-            .drain(..)
-            .chain(finalized_blocks.into_iter())
-            .collect();
-        css_final_blocks.sort_unstable_by_key(|(_, b)| b.header.content.slot);
+        // process SCE-final slots
+        self.pending_css_final_blocks
+            .extend(finalized_blocks.into_iter().filter_map(|(b_id, b)| {
```

Comment:
here I would assign the result of the `filter_map` to a variable named `new_css_final_blocks`, to match the wording of the spec.

Reply:
done